openmrs-19/src/test/java/org/motechproject/openmrs19/resource/impl/PersonResourceImplTest.java

```diff
@@ -42,6 +42,46 @@ public void shouldGetPersonById() throws HttpException, IOException {
     }
 
     @Test
+    public void shouldGetAllAdressFields() throws HttpException, IOException {
+        Mockito.when(getClient().getJson(Mockito.any(URI.class)))
+                .thenReturn(readJsonFromFile("json/person-response.json"));
+
+        String address1 = "5 Main St.";
+        String address2 = "5/4";
+        String cityVillage = "Utopia";
+        String stateProvince = "testProvince";
+        String country = "Neverland";
+        String postalCode = "69-111";
+        String countyDistrict = null;
+        String address3 = null;
+        String address4 = null;
+        String address5 = null;
+        String address6 = null;
+        String startDate = "2016-03-01T00:00:00.000+0000";
+        String endDate = null;
+        String latitude = "47.613879";
+        String longitude = "-122.342436";
+
+        Person person = impl.getPersonById("PPP");
+
+        assertEquals(address1, person.getPreferredAddress().getAddress1());
```

Comment:
here you are only checking json parsing, you should rather check if method getFullAddressString in Person class is properly merging all address fields into one string.